### PR TITLE
Headers are added for the different sections, merged results, filter for songs without certain instruments

### PR DESCRIPTION
**Pull request by SamuelPalma**
* Headers are only shown if the song list is greater than 30 songs (arbritrary number)
    + TODO: Tweak constant MIN_SONGS_FOR_SECTION_HEADERS

* Added filters to search for songs without a specific instrument 
    + instrument:-guitar 
    + instrument:-bass 
    + instrument:-drums 
    + instrument:-keys 
    + instrument:-vocals 
    + instrument:-band

* The search results are merged
    + source:rb1dlc;source:rb2dlc

* Fixed bug that displayed the following filter incorrectly if the back button was pressed

* SongSelection.cs class refactored
    + SongSearching.cs created

### CODE REVIEW

**Comment by SamuelPalma** [comment]
Filters for songs without a specific instrument

YARG_20230611_235706

**Comment by SamuelPalma** [comment]
Merged results
YARG_20230612_000213

**Comment by SamuelPalma** [comment]
Section headers
YARG_20230612_000601

**Comment by SamuelPalma** [comment]
Hello, the current implementation of the headers is a button without associated functionality [ButtonViewType](https://github.com/YARC-Official/YARG/blob/master/Assets/Script/UI/MusicLibrary/ViewTypes/ButtonViewType.cs) that is inserted in the appropriate place. 

Apparently the [CategoryViewType](https://github.com/YARC-Official/YARG/blob/master/Assets/Script/UI/MusicLibrary/ViewTypes/CategoryViewType.cs) class already has a section for counting, but we would have to modify the way in which the songs are added, to add them by section step by step and not globally.


Regarding the subheadings we would have to create a new class that extends [ViewType](https://github.com/YARC-Official/YARG/blob/master/Assets/Script/UI/MusicLibrary/ViewTypes/ViewType.cs) that contains a list of [SongViewType](https://github.com/YARC-Official/YARG/blob/master/Assets/Script/UI/MusicLibrary/ViewTypes/SongViewType.cs), but I believe that it is possible.

File: Assets/Script/Song/SongSearching.cs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		private readonly static SongSearching _instance = new SongSearching();
+
+		public static SongSearching Instance {
+			get
+			{
+				return _instance;
+			}
+		}
+
+		private readonly static List<Tuple<string, string>> troublesomeCharacters = new List<Tuple<string, string>>{
```
Comment (EliteAsian123) [inline Assets/Script/Song/SongSearching.cs:31]: `List<Tuple<string, string>>` can be simplified to `List<(string, string)>`, or better yet, `List<(string from, string to)>`.

**Review by EliteAsian123 — CHANGES_REQUESTED** [review]

**Review by EliteAsian123 — APPROVED** [review]

**Pull request merged**